### PR TITLE
[MISC] rename align_cfg::scoring to align_cfg::scoring_scheme

### PR DESCRIPTION
**Pull request by wvandertoorn**
Fixes https://github.com/seqan/product_backlog/issues/177.

### CODE REVIEW

File: test/performance/alignment/global_affine_alignment_benchmark.cpp [hunk 1]
```diff
@@ -29,7 +29,7 @@
 constexpr auto affine_cfg = seqan3::align_cfg::method_global{} |
                             seqan3::align_cfg::gap{seqan3::gap_scheme{seqan3::gap_score{-1},
                                                                       seqan3::gap_open_score{-10}}} |
-                            seqan3::align_cfg::scoring{seqan3::nucleotide_scoring_scheme{seqan3::match_score{4},
+                            seqan3::align_cfg::scoring_scheme{seqan3::nucleotide_scoring_scheme{seqan3::match_score{4},
```
Comment (joergi-w) [inline test/performance/alignment/global_affine_alignment_benchmark.cpp:32]: adapt the indentation of the line below accordingly, please

File: test/performance/alignment/global_affine_alignment_parallel_benchmark.cpp [hunk 2]
```diff
@@ -34,7 +34,7 @@
 constexpr auto affine_cfg = seqan3::align_cfg::method_global{} |
                             seqan3::align_cfg::gap{seqan3::gap_scheme{seqan3::gap_score{-1},
                                                                       seqan3::gap_open_score{-10}}} |
-                            seqan3::align_cfg::scoring{seqan3::nucleotide_scoring_scheme{seqan3::match_score{4},
+                            seqan3::align_cfg::scoring_scheme{seqan3::nucleotide_scoring_scheme{seqan3::match_score{4},
```
Comment (joergi-w) [inline test/performance/alignment/global_affine_alignment_parallel_benchmark.cpp:37]: same here

File: test/performance/alignment/global_affine_alignment_simd_benchmark.cpp [hunk 3]
```diff
@@ -29,7 +29,7 @@
 constexpr auto affine_cfg = seqan3::align_cfg::method_global{} |
                             seqan3::align_cfg::gap{seqan3::gap_scheme{seqan3::gap_score{-1},
                                                                       seqan3::gap_open_score{-10}}} |
-                            seqan3::align_cfg::scoring{seqan3::nucleotide_scoring_scheme{seqan3::match_score{4},
+                            seqan3::align_cfg::scoring_scheme{seqan3::nucleotide_scoring_scheme{seqan3::match_score{4},
```
Comment (joergi-w) [inline test/performance/alignment/global_affine_alignment_simd_benchmark.cpp:32]: and here

File: test/performance/io/format_sam_benchmark.cpp [hunk 4]
```diff
@@ -39,7 +39,7 @@ static std::string create_sam_file_string(size_t const n_queries)
 
         // align
         auto config = seqan3::align_cfg::method_global{} |
-                      seqan3::align_cfg::scoring{seqan3::nucleotide_scoring_scheme{seqan3::match_score{4},
+                      seqan3::align_cfg::scoring_scheme{seqan3::nucleotide_scoring_scheme{seqan3::match_score{4},
```
Comment (joergi-w) [inline test/performance/io/format_sam_benchmark.cpp:42]: also here

File: test/unit/alignment/pairwise/fixture/global_affine_banded.hpp [hunk 5]
```diff
@@ -50,7 +50,7 @@ static auto dna4_01 = []()
         "ACGTACGTA"_dna4,
         align_config | seqan3::align_cfg::band_fixed_size{seqan3::align_cfg::lower_diagonal{-3},
                                                           seqan3::align_cfg::upper_diagonal{8}} |
-                       seqan3::align_cfg::scoring{seqan3::nucleotide_scoring_scheme{seqan3::match_score{4},
+                       seqan3::align_cfg::scoring_scheme{seqan3::nucleotide_scoring_scheme{seqan3::match_score{4},
```
Comment (joergi-w) [inline test/unit/alignment/pairwise/fixture/global_affine_banded.hpp:53]: Please adapt the indentation of the following line accordingly (+5x below).

File: test/unit/alignment/pairwise/fixture/local_affine_banded.hpp [hunk 6]
```diff
@@ -46,7 +46,7 @@ static auto dna4_01 = []()
         // GTCTA
         "AACCGGTTTAACCGGTT"_dna4,
         "ACGTCTACGTA"_dna4,
-        align_config | seqan3::align_cfg::scoring{seqan3::nucleotide_scoring_scheme{seqan3::match_score{4},
+        align_config | seqan3::align_cfg::scoring_scheme{seqan3::nucleotide_scoring_scheme{seqan3::match_score{4},
```
Comment (joergi-w) [inline test/unit/alignment/pairwise/fixture/local_affine_banded.hpp:49]: Please adapt the indentation of the following line accordingly (+5x below).

File: test/unit/alignment/pairwise/fixture/local_affine_unbanded.hpp [hunk 7]
```diff
@@ -45,7 +45,7 @@ static auto dna4_01 = []()
         // GTCTA
         "AACCGGTTTAACCGGTT"_dna4,
         "ACGTCTACGTA"_dna4,
-        align_config | seqan3::align_cfg::scoring{seqan3::nucleotide_scoring_scheme{seqan3::match_score{4},
+        align_config | seqan3::align_cfg::scoring_scheme{seqan3::nucleotide_scoring_scheme{seqan3::match_score{4},
```
Comment (joergi-w) [inline test/unit/alignment/pairwise/fixture/local_affine_unbanded.hpp:48]: Please adapt the indentation of the following line accordingly (+5x below).

File: test/unit/alignment/pairwise/fixture/semi_global_affine_banded.hpp [hunk 8]
```diff
@@ -43,7 +43,7 @@ static auto dna4_01_semi_first = []()
         "TTTTTACGTATGTCCCCC"_dna4,
         "ACGTAAAACGT"_dna4,
         align_config_semi_seq1
-            | seqan3::align_cfg::scoring{seqan3::nucleotide_scoring_scheme{seqan3::match_score{4},
+            | seqan3::align_cfg::scoring_scheme{seqan3::nucleotide_scoring_scheme{seqan3::match_score{4},
```
Comment (joergi-w) [inline test/unit/alignment/pairwise/fixture/semi_global_affine_banded.hpp:46]: Please adapt the indentation of the following line accordingly (+2x below).

File: test/unit/alignment/pairwise/fixture/semi_global_affine_unbanded.hpp [hunk 9]
```diff
@@ -43,7 +43,7 @@ static auto dna4_01_semi_first = []()
         "TTTTTACGTATGTCCCCC"_dna4,
         "ACGTAAAACGT"_dna4,
         align_config_semi_seq1
-            | seqan3::align_cfg::scoring{seqan3::nucleotide_scoring_scheme{seqan3::match_score{4},
+            | seqan3::align_cfg::scoring_scheme{seqan3::nucleotide_scoring_scheme{seqan3::match_score{4},
```
Comment (joergi-w) [inline test/unit/alignment/pairwise/fixture/semi_global_affine_unbanded.hpp:46]: Please adapt the indentation of the following line accordingly (+3x below).

File: test/unit/alignment/pairwise/semi_global_affine_banded_test.cpp [hunk 10]
```diff
@@ -35,7 +35,7 @@ struct pairwise_semiglobal_affine_banded : public ::testing::Test
         return seqan3::align_cfg::method_global{} |
                seqan3::align_cfg::gap{seqan3::gap_scheme{seqan3::gap_score{-1},
                                                          seqan3::gap_open_score{-10}}} |
-               seqan3::align_cfg::scoring{seqan3::nucleotide_scoring_scheme{seqan3::match_score{4},
+               seqan3::align_cfg::scoring_scheme{seqan3::nucleotide_scoring_scheme{seqan3::match_score{4},
```
Comment (joergi-w) [inline test/unit/alignment/pairwise/semi_global_affine_banded_test.cpp:38]: Please adapt the indentation of the following line accordingly.

**Comment by wvandertoorn** [comment]
@joergi-w thank you for pointing all the files out, I'm sorry to have overlooked the indentation shifts.
As for the naming concerns, that is a good observation. @seqan/core

**Comment by rrahn** [comment]
We need to rename the `scoring_scheme` concept to `scoring_scheme_for` in order to reflect the concept naming rules of the standard. The rational is that scoring_scheme_for requires an additional type to check for. So I think we can keep the align_cfg naming as is.

**Comment by wvandertoorn** [comment]
Rebased as requested to avoid some mess with the align_cfg::detail namespace (https://github.com/seqan/seqan3/pull/2020), and cleaned up the commit history.

**Review by smehringer — APPROVED** [review]
LGTM :)

**Pull request merged**